tantalus/models.py

```diff
@@ -978,4 +980,111 @@ def get_absolute_url(self):
         return reverse("submission-detail", args=(self.id,))
 
     def __str__(self):
-        return str(self.sample.sample_id)
\ No newline at end of file
+        return str(self.sample.sample_id)
+
+
+class Curation(models.Model):
+    """
+    Simple text curation associated with datasets.
+    """
+
+    history = HistoricalRecords()
+
+    name = create_id_field(unique=True)
+
+    owner = models.ForeignKey(
+        account.models.User,
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    sequencedatasets = models.ManyToManyField(
+        SequenceDataset,
+        blank=True
+    )
+    description = models.CharField(
+        max_length=500,
+        null=True
+    )
+    updated = models.DateTimeField(
+        auto_now=True,
+        auto_now_add=False)
+    created = models.DateTimeField(
+        auto_now=False,
+        auto_now_add=True,
+        blank=True)
+
+    def __str__(self):
+        return self.name
+
+    def count_sequence_datasets(self):
+        '''
+        Return the number of sequence datasets that are associated with this curation.
+        '''
+        return len(self.sequencedatasets.all())
+
+    def get_absolute_url(self):
+        return reverse("curation-detail", args=(self.id,))
+
+    def get_created_time(self):
+        '''
+        Return the creation date of the curation.
+        '''
+        if self.created:
+            return self.created.strftime("%Y-%m-%d %H:%M:%S")
+        else:
+            return "Not Provided"
+    def get_last_modified_time(self):
+        '''
+        Return the latest modification time of the curation.
+        '''
+        return self.updated.strftime("%Y-%m-%d %H:%M:%S")
+    def get_data(self):
+        '''
+        Return the data contained in the curation object.
+        '''
+        data = {
+            "name": self.name,
+            "owner": str(self.owner),
+            "description": self.description,
+            "sequencedatasets": [ds.pk for ds in self.sequencedatasets.all()]
+            }
+        return data
+
+# Validator for curation version
+curation_version_validator = RegexValidator(
+    regex=r"v\d+\.\d+\.\d+",
+    message=' must be in "v<MAJOR>.<MINOR>.<PATCH>"; for example, "v0.0.1"',
+)
+
+
+class CurationHistory(models.Model):
```

Comment:
we already have automatic history tracking with django-simple-history. would be worth checking if it has the features you need as it would be less code you would have to write and maintain